### PR TITLE
Npe2 samples

### PR DESCRIPTION
**Pull request by tlambert03**
# Description
adds sample data support for npe2 ... with https://github.com/tlambert03/npe2/pull/31

### CODE REVIEW

File: napari/builtins.yaml [hunk 1]
```diff
@@ -7,57 +7,336 @@ private: true
...
+
+
+
+
+
+
+
+
+
+
```
Comment (sofroniewn) [inline napari/builtins.yaml]: Seems like there are lots of blank lines at end of file are these needed?

**Review by sofroniewn — APPROVED** [review]
Love it, I love the manifest. Really easy to follow. Very clear to me whats going on (at least i think) even though I havn't been in the weeds of this stuff lately. Nice work here!

File: napari/_qt/menus/file_menu.py [hunk 2]
```diff
@@ -160,7 +162,31 @@ def _rebuild_samples_menu(self):
 
         self.open_sample_menu.clear()
 
-        for plugin_name, samples in plugin_manager._sample_data.items():
+        # eg ('dock', ('my_plugin', {'My widget': MyWidget}))
```
Comment (nclack) [inline napari/_qt/menus/file_menu.py:165]:
```suggestion
        # eg ('plugin_name', ('sample.key', {...}))
```

File: napari/_qt/menus/file_menu.py [hunk 3]
```diff
@@ -160,7 +162,31 @@ def _rebuild_samples_menu(self):
...
+                    plugin_name,
+                    {
+                        c.key: {
+                            'data': c.open,
+                            'display_name': c.display_name,
+                        }
+                        for c in contribs
+                    },
+                )
+                for plugin_name, contribs in pm.iter_sample_data()
```
Comment (nclack) [inline napari/_qt/menus/file_menu.py:184]: `plugin_name` is getting used below to create the menu for a multiprovider.  How do you feel about using the plugin display name instead?

File: napari/builtins.yaml [hunk 4]
```diff
@@ -7,57 +7,336 @@ private: true
...
+          ".stk",
+          ".tga",
+        ]
+      layer_types: ["image"]
+      name: lossless
+
+    - command: napari_builtins.write_image
+      filename_extensions:
+        [
+          ".j2c",
```
Comment (nclack) [inline napari/builtins.yaml:169]: Would you move .jpg to the front of the list.  The first entry in this list ends up being the default selection when a user clicks on the entry in the file save dialog; j2c shouldn't be the default.  Maybe I should make a separate pr, but since it's getting changed now...

**Review by nclack — APPROVED** [review]
I left some comments.  The biggest question for me is the menu naming for multi-providers; the plugin_name is used rather than the display name. plugin_name isn't as pretty, but it is unique.

File: napari/builtins.yaml (hunk 4 above)
Comment (tlambert03) [reply napari/builtins.yaml:169]: sorry, this was all just autoformatting... will change back

Comment (tlambert03) [reply napari/builtins.yaml:169]: > Would you move .jpg to the front of the list.

sure though :)

Comment (nclack) [reply napari/builtins.yaml:169]: Thanks!  I keep fighting w the autoformatter here too.  I'm happy letting it win this one :)

File: napari/plugins/_skimage_data.py [hunk 5]
```diff
@@ -83,11 +83,19 @@ def _load_skimage_data(name, **kwargs):
     return [(getattr(skimage.data, name)(**kwargs), {'name': name})]
 
 
-@napari_hook_implementation
-def napari_provide_sample_data():
-    from functools import partial
+_DATA = {
+    key: {'data': partial(_load_skimage_data, key), 'display_name': dname}
```
Comment (goanpeca) [inline napari/plugins/_skimage_data.py:87]: @tlambert03 could we wrap the display name with a `trans._(dname)` call?

We have not updated the translation parsing for the manifest files (yet) but this is a good reminder :)

File: napari/_qt/menus/file_menu.py [hunk 6]
```diff
@@ -160,7 +162,31 @@ def _rebuild_samples_menu(self):
...
+            _iterable = iter([])
+        else:
+            pm = npe2.PluginManager.instance()
+            _iterable = (
+                (
+                    plugin_name,
+                    {
+                        c.key: {
+                            'data': c.open,
+                            'display_name': c.display_name,
```
Comment (goanpeca) [inline napari/_qt/menus/file_menu.py:179]: @tlambert03 could we wrap the display name with a `trans._(dname)` call?

We have not updated the translation parsing for the manifest files (yet) but this is a good reminder :)

File: napari/plugins/_skimage_data.py (hunk 5 above)
Comment (tlambert03) [reply napari/plugins/_skimage_data.py:87]: these are already wrapped in `trans` above in the `SKIMAGE_DATA` declaration

File: napari/builtins.yaml (hunk 1 above)
Comment (tlambert03) [reply napari/builtins.yaml]: no :joy:

File: napari/_qt/menus/file_menu.py (hunk 2 above)
Comment (tlambert03) [reply napari/_qt/menus/file_menu.py:165]: this is getting moved anyway in #3666

File: napari/_qt/menus/file_menu.py (hunk 6 above)
Comment (tlambert03) [reply napari/_qt/menus/file_menu.py:179]: help me understand the flow here.  Since this `display_name` is coming from the manifest, would that suggest that we need to use `trans._` _everywhere_ we pull text from the manifest?  I would have assumed we could do that conversion somewhere higher up in the chain, rather than at the point of usage of the string

File: napari/_qt/menus/file_menu.py (hunk 3 above)
Comment (tlambert03) [reply napari/_qt/menus/file_menu.py:184]: don't mind that, but maybe let's deal with that elsewhere?  The behavior here is consistent with npe1... it would be nice to keep the menus consistent somehow.

File: napari/_qt/menus/file_menu.py (hunk 6 above)
Comment (goanpeca) [reply napari/_qt/menus/file_menu.py:179]: >conversion somewhere higher up in the chain, rather than at the point of usage of the string

Well yes, we could do that wherever we parse the manifest then. We just need to to it somehwere :)

Comment (tlambert03) [reply napari/_qt/menus/file_menu.py:179]: yeah, lets do that :)

Comment (goanpeca) [reply napari/_qt/menus/file_menu.py:179]: Sounds good!
